src/react-components/src/__tests__/components/Link-test.tsx

```diff
@@ -0,0 +1,98 @@
+import { expect } from "chai";
+import { stub } from "sinon";
+import * as Enzyme from "enzyme";
+import * as React from "react";
+import * as Mocha from "mocha";
+
+import Link from "../../components/Link/Link";
+import Icon from "../../components/Icons/Icon";
+
+describe("Link", () => {
+    let wrapper: Enzyme.ReactWrapper<{}, {}>;
+
+    it("Can pass in icon, text as child and url as props", () => {
+        wrapper = Enzyme.mount(
+            <Link href="#test">
+                Test
+                <Icon
+                    name="arrow_xsmall"
+                    blockName="more-link"
+                    modifiers={["right"]}
+                    decorative={true}
+                ></Icon>
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+        expect(wrapper.find("Icon")).to.have.lengthOf(1);
+    });
+
+    it("Can pass a link with <a> tag", () => {
+        wrapper = Enzyme.mount(
+            <Link>
+                <a href="#test">Test</a>
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+    });
+
+    it("Can pass a link with <a> tag and an icon", () => {
+        wrapper = Enzyme.mount(
+            <Link>
+                <a href="#test">Test</a>
+                <Icon
+                    name="arrow_xsmall"
+                    blockName="more-link"
+                    modifiers={["right"]}
+                    decorative={true}
+                ></Icon>
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+        expect(wrapper.find("Icon")).to.have.lengthOf(1);
+    });
+
+    it("Can pass in left icon", () => {
+        wrapper = Enzyme.mount(
+            <Link href="#nice-url">
+                <Icon
+                    name="arrow_xsmall"
+                    blockName="more-link"
+                    modifiers={["left"]}
+                    decorative={true}
+                ></Icon>
+                content
+            </Link>
+        );
+        expect(
+            wrapper.find(".more-link__icon").hasClass("more-link__icon--left")
+        ).to.equal(true);
+    });
+
+    it("Can pass in right icon", () => {
+        wrapper = Enzyme.mount(
+            <Link href="#whatever">
+                content
+                <Icon
```

Comment:
So this (and all the stories) are wrapping the icon rather than passing in an icon (using `iconPosition` and `iconRotation`).  Should `iconPosition` and `iconRotation` be used at all in `Link` or should it always be passed in like this?

Reply:
Since I've added the other linkTypes I think the consumer shouldn't have to pass these in to Link, so I've updated this.